pkg/kubelet/dockertools/labels.go

```diff
@@ -34,20 +35,28 @@ import (
 //  * label filters (maybe in the future)
 
 const (
-	kubernetesPodNameLabel                   = "io.kubernetes.pod.name"
-	kubernetesPodNamespaceLabel              = "io.kubernetes.pod.namespace"
-	kubernetesPodUIDLabel                    = "io.kubernetes.pod.uid"
-	kubernetesPodDeletionGracePeriodLabel    = "io.kubernetes.pod.deletionGracePeriod"
-	kubernetesPodTerminationGracePeriodLabel = "io.kubernetes.pod.terminationGracePeriod"
-
-	kubernetesContainerNameLabel                   = "io.kubernetes.container.name"
-	kubernetesContainerHashLabel                   = "io.kubernetes.container.hash"
-	kubernetesContainerRestartCountLabel           = "io.kubernetes.container.restartCount"
-	kubernetesContainerTerminationMessagePathLabel = "io.kubernetes.container.terminationMessagePath"
-	kubernetesContainerPreStopHandlerLabel         = "io.kubernetes.container.preStopHandler"
+	podNameLabel                   = "io.kubernetes.pod.name"
+	podNamespaceLabel              = "io.kubernetes.pod.namespace"
+	podUIDLabel                    = "io.kubernetes.pod.uid"
+	podDeletionGracePeriodLabel    = "io.kubernetes.pod.deletion-grace-period"
+	podTerminationGracePeriodLabel = "io.kubernetes.pod.termination-grace-period"
+
+	containerNameLabel                   = "io.kubernetes.container.name"
+	containerHashLabel                   = "io.kubernetes.container.hash"
+	containerRestartCountLabel           = "io.kubernetes.container.restart-count"
+	containerTerminationMessagePathLabel = "io.kubernetes.container.termination-message-path"
+	containerPreStopHandlerLabel         = "io.kubernetes.container.prestop-handler"
+
+	// TODO(random-liu): These labels don't follow the docker label naming convention, deprecate this
+	// when we drop support for v1.2.
+	oldPodDeletionGracePeriodLabel          = "io.kubernetes.pod.deletionGracePeriod"
+	oldPodTerminationGracePeriodLabel       = "io.kubernetes.pod.terminationGracePeriod"
+	oldContainerRestartCountLabel           = "io.kubernetes.container.restartCount"
+	oldContainerTerminationMessagePathLabel = "io.kubernetes.container.terminationMessagePath"
+	oldContainerPreStopHandlerLabel         = "io.kubernetes.container.preStopHandler"
```

Comment:
I'd recommend grouping all the label backwards compat logic here in this file, close to the label declarations..
I see this PR being similar to how we handle backwards compatibility with the REST API.